pkg/auth/credentials.go

```diff
@@ -199,3 +199,15 @@ func CreateCredentials(accessKey, secretKey string) (cred Credentials, err error
 	cred.Status = "enabled"
 	return cred, nil
 }
+
+// CreateCredentialsWithMetadata returns new credentials with
+// given access key, secret key, session token and expiry
+func CreateCredentialsWithMetadata(accessKey, secretKey, sessionToken string, exipresAt time.Time) (cred Credentials, err error) {
```

Comment:
The function name is not correct @kanagarajkm you just have use the creds as is there is not reason for this function to exist. 

authenticateWebSTS() should directly take a.Result.Credentials

Reply:
@harshavardhana will remove this function, thanks!